staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -972,6 +972,15 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 		return false, nil, nil
 	}
 
+	// Since adminAccess allocations are not tracked, check again that the
+	// device hasn't already been allocated for this claim.
+	if request.adminAccess() && slices.ContainsFunc(alloc.result[r.claimIndex].devices, func(previousResult internalDeviceResult) bool {
```

Comment:
Could we use `allocatingDevices` here? I think it currently tracks all devices being allocated across all claims, so it would need to keep track of allocating devices per claim. But if it did, it would be used to check both if the device is being allocated by the "current" claim (for admin access) or by any claim (for the regular check).

But the cost of scanning the list of devices in the result is probably not significant here.

Reply:
Yeah, it seems like that works and I think I like that better. I pushed that change but I'll plan to run both versions through scheduler_perf tomorrow to see if there are any surprising results.